### DISCUSSION TITLE
Merge history after deleting a primary e-mail

### DISCUSSION THREAD

**Discussion by janeman98** [community/community#23260] [New to GitHub]
<p>Hi,</p>
<p>Currently, I have emailA as primary and emailB as backup. Now, I want to set emailB as primary, but still want to keep all the merge history using emailA to associate with my account. Is there a way to do so? When I try delete my primary email (emailA), I got a pop saying:</p>
<p>“Are you sure you want to remove this email from your account? Once removed, commits attributed to this email address will no longer be associated with your account. One of your other emails will become your primary address.”</p>
<p>Does this mean all merge history using emailA will no longer associated with my account? Maybe, is there a way to swap the primary and backup e-mail?</p>
<p>Thanks</p>

**Comment by janeman98** [comment, marked as answer]
<p>This can be closed.</p>
<p>I need to add a 3rd email before GitHub allow me to swap the primary and backup email.</p>

**Reply by MaxChandlerII** [reply]
Hi @janeman98, I'm in a similar situation. I have emailA that I want to delete, but it has commits associated with it. I have emailB that was backup. I added a 3rd email to GitHub and made the 3rd email my primary. However, even after adding the 3rd email I'm unabled to delete emailA without receiving the same popup you saw. How did you resolve that issue?
